common.go

```diff
@@ -30,7 +30,7 @@ const (
 	DOCKER = "docker"
 )
 
-func getCopyOptions(store storage.Store, reportWriter io.Writer, sourceSystemContext *types.SystemContext, destinationSystemContext *types.SystemContext, manifestType string, removeSignatures bool, addSigner string) *cp.Options {
+func getCopyOptions(store storage.Store, reportWriter io.Writer, sourceSystemContext *types.SystemContext, destinationSystemContext *types.SystemContext, manifestType string, removeSignatures bool, addSigner string, pullAlways bool) *cp.Options {
```

Comment:
This function has gotten pretty ridiculous.  I don't see much value in it as it is only populating a structure that you have to pass in all of the options.  Perhaps two functions to get sourceCtx and DestinationCtx?